### DISCUSSION TITLE
Force the tooltip show during a programmed automatic demonstration

### DISCUSSION THREAD

**Discussion by kaz-yos** [company-mode/company-mode#1174] [Q&A] [answered]
Thanks for maintaining a great package!

I short, I want the following non-interactive code to show the tooltip for three seconds before moving on to complete with the first candidate.

```lisp
(progn
  (company-auto-begin)
  (sit-for 3)
  (company-complete-selection))
```
If evaluated in the elisp mode after typing `company`, C-g (to suppress regular company mode auto start), M-: (following) expression, it sits for 3 seconds showing nothing and then `company--auto-completion` shows up.

```lisp
(progn
  (company-auto-begin)
  (sit-for 3))
```
Evaluating this will show nothing for 3 seconds and then bring up the tooltip.

```lisp
(company-auto-begin)
```
Evaluating this will show the tooltip immediately.

I am trying to fully automate the demonstration of `stan-mode` packages like the following.

![example](https://user-images.githubusercontent.com/2884877/75091986-4198f580-5541-11ea-8df6-89045748cd5c.gif)

**Comment by dgutov** [comment, marked as answer]
Hi!

Try this one:

```lisp
(progn
  (company-manual-begin)
  (let ((this-command 'self-insert-command))
    (company-post-command))
  (sit-for 3)
  (company-complete-selection))
```

**Comment by kaz-yos** [comment]
This works like a charm!! Thank you so much!
